### ISSUE TITLE
Bardess Qlik Sense App

### ISSUE THREAD

**Issue by KPMGIndia** [bardess/ios-webkit-wrapper#1]
To avoid websocket problem on mobile iron, we were suggested using Bardess mobile qliksense app. 
http://branch.qlik.com/#!/project/58f4de153cdc2fefcfa1aa63 

however, it still shows the hub loading screen and doesn't render the application or the hub. 
The login page appeared correctly. 
The logs show the last URL to be loaded correctly. 
![photo-1](https://user-images.githubusercontent.com/29978572/27957145-a2e01022-633a-11e7-9692-dcaa0095b259.jpg)
![photo-2](https://user-images.githubusercontent.com/29978572/27957144-a2c68bf2-633a-11e7-98d0-25f5fa91d7fa.jpg)

**Comment by jamesfleming** [comment]
what url? is it the default url for demos.bardesscloud.com or an internal server url? what Qlik Sense version?

**Comment by KPMGIndia** [comment]
Hello !

Qliksense version - 3.0

I am giving internal server URL where my Qliksense is hosted.

We put in our Qliksense Hub URL and then built the project using Xcode.
Deployed the IPA through mobile iron.

Warm Regards,

Kashif Shah
Manager – Technology
KPMG in India

Contact: +91 124 334 5044  | M: + 91 7702778896 | VoIP:
Email: kashifshah@kpmg.com
Gurgaon
[cid:image001.png@01D2F75B.59AEACB0]

**Comment by jamesfleming** [comment]
tough to say what the issues is without a stack trace, we implemented a way to do that ourselves by emailing it to someone. you can see the code for this in /WebkitWraper/WebViewViewController.swift sendToMail function on line 143. let me know what you get out of that

**Comment by KPMGIndia** [comment]
All right, will try this out very soon and let you know .

Thank you for your prompt response !

Sent from Email+ secured by MobileIron

-------- Original Message --------

**Comment by KPMGIndia** [comment]
Dear James,

We have the bardess app running now, after upgrading to release 3.2.4
It loads the hub screen as I have set it to default URL.
However, when I click on any app, it keeps loading the page and nothing happens.

We have whitelisted the bardess app in mobile iron to send mail via Email+ but this isn't working too, I haven't been able to do much about this.

Could you advise from experience ?

Warm Regards,

Kashif Shah
Manager - Technology
KPMG in India

Contact: +91 124 334 5044  | M: + 91 7702778896 | VoIP:
Email: kashifshah@kpmg.com
Gurgaon
[cid:image001.png@01D30543.7B2D7240]

**Comment by jamesfleming** [comment]
If you can load the hub (which is mainly static web assets) but not an app it appears as if you are having issues with the websocket. My suggestion based on past experience is to test the websocket connectivity extension and see what result you get: http://branch.qlik.com/#!/project/56728f52d1e497241ae69865

Also take a look at the error log in that email to see what if anything is popping up in the browser console.

**Comment by KPMGIndia** [comment]
Dear James,

I rebuilt the soln and it is working now.
Though at times a bit sketchy giving me NSURL errors, I am testing it thoroughly and will update you soon.

Warm Regards,

Kashif Shah
Manager – Technology
KPMG in India

Contact: +91 124 334 5044  | M: + 91 7702778896 | VoIP:
Email: kashifshah@kpmg.com
Gurgaon
[cid:image001.png@01D30ABE.13B995F0]

**Comment by KPMGIndia** [comment]
Dear James,

Trust you are doing well.

I was able to get the email up and running. And after a couple of trials, was able to login and see the hub again.

The app keeps loading when I click on a dashboard, attaching the log for the same.
Surprisingly, the Qlik web socket connectivity page shows websocket error. Is this expected through Bardess ?


Warm Regards,

Kashif Shah
Manager – Technology
KPMG in India

Contact: +91 124 334 5044  | M: + 91 7702778896 | VoIP:
Email: kashifshah@kpmg.com<mailto:kashifshah@kpmg.com>
Gurgaon
[cid:image001.png@01D31071.42E9BE00]

**Comment by jamesfleming** [comment]
Yes we have seen that web socket connectivity issue in the past in other environments and it is often the key blocker of this implementation. There isn’t a simple answer to what causes that because it can be a lot of things depending on the environment.
